google-datacatalog-qlik-connector/src/google/datacatalog_connectors/qlik/prepare/assembled_entry_factory.py

```diff
@@ -26,69 +26,100 @@ class AssembledEntryFactory:
 
     def __init__(self, project_id, location_id, entry_group_id,
                  user_specified_system, site_url):
+
         self.__datacatalog_entry_factory = datacatalog_entry_factory \
             .DataCatalogEntryFactory(
                 project_id, location_id, entry_group_id, user_specified_system,
                 site_url)
 
+        self.__app_tag_template = None
+        self.__sheet_tag_template = None
+        self.__stream_tag_template = None
+
         self.__datacatalog_tag_factory = \
             datacatalog_tag_factory.DataCatalogTagFactory(site_url)
 
     def make_assembled_entries_list(self, stream_metadata, tag_templates_dict):
-        stream_tag_template = self.__get_tag_template(
-            constants.TAG_TEMPLATE_ID_STREAM, tag_templates_dict)
+        self.__initialize_tag_templates(tag_templates_dict)
 
         assembled_entries = [
-            self.__make_assembled_entry_for_stream(stream_metadata,
-                                                   stream_tag_template)
+            self.__make_assembled_entry_for_stream(stream_metadata)
         ]
 
-        app_tag_template = self.__get_tag_template(
-            constants.TAG_TEMPLATE_ID_APP, tag_templates_dict)
-
         assembled_entries.extend(
-            self.__make_assembled_entries_for_apps(stream_metadata,
-                                                   app_tag_template))
+            self.__make_assembled_entries_for_apps(
+                stream_metadata.get('apps')))
 
         return assembled_entries
 
+    def __initialize_tag_templates(self, tag_templates_dict):
```

Comment:
> What's the benefit of this method __initialize_tag_templates ?

Just to handle the templates initialization logic apart.

> Looking at the code flow, this seems to always resolve to true, any scenario where it could be false?

The templates are created by the caller (usually the synchronizer class) and received by the public `make_assembled_entries_list` method as an input arg. If for any reason the code is changed upstream and any of the expected templates is not received anymore, it resolves to false.